dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/XmlTool.java

```diff
@@ -146,8 +149,17 @@ protected void read(String file) throws Exception {
 	 * this instance.
 	 */
 	protected void read(URL url) throws Exception {
-		SAXReader reader = new SAXReader();
-		setRoot(reader.read(url));
+	    
+	    
+	    if(IPUtils.isIpPrivateSubnet(url.getHost()) && !Config.getBooleanProperty("ALLOW_ACCESS_TO_PRIVATE_SUBNETS", false)){

```

Comment:
maybe we should use a util class for this logic, maybe in IPUtils

Reply:
probably in `IPUtils`?

Reply:
I would encapsulate the Config call in a Lazy